apps/cyberstorm-remix/cyberstorm/utils/StrongForm/useStrongForm.ts

```diff
@@ -3,31 +3,105 @@ import {
   ParseError,
   RequestBodyParseError,
   RequestQueryParamsParseError,
-} from "../../../../../packages/thunderstore-api/src";
+  UserFacingError,
+  mapApiErrorToUserFacingError,
+} from "@thunderstore/thunderstore-api";
 
-interface UseStrongFormProps<
-  Inputs,
+type IsExact<A, B> = (<T>() => T extends A ? 1 : 2) extends <T>() => T extends B
```

Comment:
Now here I would welcome a good docstring explaining `IsExact`, `RefinerRequirement`, and `ErrorMapperRequirement`.